### PR TITLE
Update to SymbolicUtils v2

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Fixes https://github.com/SciML/ModelingToolkit.jl/issues/2753

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
```julia
  LoadError: eachindex of x(t) not known
  Stacktrace:
    [1] error(s::String)
      @ Base ./error.jl:35
    [2] eachindex(A::SymbolicUtils.BasicSymbolic{Matrix{Real}})
      @ Symbolics ~/.julia/packages/Symbolics/D9Cyv/src/arrays.jl:571
    [3] ModelingToolkit.IndexCache(sys::ModelingToolkit.ODESystem)
      @ ModelingToolkit ~/work/ModelingToolkit.jl/ModelingToolkit.jl/src/systems/index_cache.jl:61
```

@shashi do you know what happened to eachindex of ArraySymbolic?

**Comment by Vaibhavdixit02** [comment]
@ChrisRackauckas @shashi is there an ETA for this?

**Comment by ChrisRackauckas** [comment]
ASAP. I'm trying to find @shashi to help me finish it 😅

File: src/systems/abstractsystem.jl [hunk 1]
```diff
@@ -2564,8 +2565,8 @@ is_diff_equation(eq2) # false
 """
 function is_diff_equation(eq)
     (eq isa Equation) || (return false)
-    isdefined(eq, :lhs) && occursin(is_derivative, wrap(eq.lhs)) && (return true)
-    isdefined(eq, :rhs) && occursin(is_derivative, wrap(eq.rhs)) && (return true)
+    isdefined(eq, :lhs) && hasnode(is_derivative, wrap(eq.lhs)) && (return true)
+    isdefined(eq, :rhs) && hasnode(is_derivative, wrap(eq.rhs)) && (return true)
```
Comment (ChrisRackauckas) [inline src/systems/abstractsystem.jl:2569]:
```suggestion
    isdefined(eq, :lhs) && SymbolicUtils.hasnode(is_derivative, wrap(eq.lhs)) && (return true)
    isdefined(eq, :rhs) && SymbolicUtils.hasnode(is_derivative, wrap(eq.rhs)) && (return true)
```

File: src/systems/abstractsystem.jl [hunk 2]
```diff
@@ -2564,8 +2565,10 @@ is_diff_equation(eq2) # false
 """
 function is_diff_equation(eq)
     (eq isa Equation) || (return false)
-    isdefined(eq, :lhs) && occursin(is_derivative, wrap(eq.lhs)) && (return true)
-    isdefined(eq, :rhs) && occursin(is_derivative, wrap(eq.rhs)) && (return true)
+    isdefined(eq, :lhs) && SymbolicUtils.hasnode(is_derivative, wrap(eq.lhs)) &&
+        (return true)
+    isdefined(eq, :rhs) && SymbolicUtils.hasnode(is_derivative, wrap(eq.rhs)) &&
+        (return true)
```
Comment (ChrisRackauckas) [inline src/systems/abstractsystem.jl]:
```suggestion
    isdefined(eq, :lhs) && hasnode(is_derivative, wrap(eq.lhs)) &&
        (return true)
    isdefined(eq, :rhs) && hasnode(is_derivative, wrap(eq.rhs)) &&
        (return true)
```

**Pull request merged**